library/src/blas3/rocblas_trmm.hpp

```diff
@@ -0,0 +1,1488 @@
+                                                     m,
+                                                     n,
+                                                     0,
+                                                     &zero,
+                                                     0,
+                                                     c,
+                                                     0,
+                                                     lda > ldc ? lda : ldc,
+                                                     0,
+                                                     c,
+                                                     0,
+                                                     lda > ldc ? lda : ldc,
+                                                     0,
+                                                     &zero,
+                                                     0,
+                                                     c,
+                                                     0,
+                                                     ldc,
+                                                     0,
+                                                     1)));
+            return rocblas_status_success;
+        }
+
+        // grid size for rocblas_copy_template
+        constexpr rocblas_int NB = 256;
+
+        // assign space for dt1 and dt2
+        rocblas_int rb = RB, cb = CB;
+        rocblas_int ldt1 = rb, ldt2 = cb;
+        T*          dt1 = workspace;
+        T*          dt2 = workspace + rb * cb;
+
+        rocblas_int    offd = rocblas_diagonal_unit == diag ? 1 : 0;
+        rocblas_int    isec, jsec, tsec;
+        T              delta;
+        rocblas_status status = rocblas_status_success;
+
+        if(side == rocblas_side_left)
+        {
+            if(uplo == rocblas_fill_upper)
+            {
+                if(transa == rocblas_operation_none)
+                {
+                    //
+                    //              Form  C := alpha*A*C. Left, Upper, No transpose.
+                    //
+                    delta     = *alpha;
+                    bool cldc = dcld(ldc);
+                    for(int ii = 1; ii <= m; ii += cb)
+                    {
+                        isec = cb < m - ii + 1 ? cb : m - ii + 1;
+                        //
+                        //                  T2 := A', the transpose of a upper unit or non-unit
+                        //                  triangular diagonal block of A is copied to the
+                        //                  lower triangular part of T2.
+                        //
+                        for(int i = ii + offd; i <= ii + isec - 1; i++)
+                        {
+                            CHECK_INTERNAL_ROCBLAS_ERROR(
+                                (rocblas_copy_template<NB, T>(handle,
+                                                              i - ii + 1 - offd,
```

Comment:
Parentheses are only needed around `rocblas_copy_template<NB, T>`, not the whole call. Makes `clang-format` work better sometimes.

Also, `T` does not need be passed as a template argument, since it can be automatically deduced; hence, `<NB, T>` can be replaced with `<NB>`, and there is no need for extra parentheses.


`CHECK_INTERNAL_ROCBLAS_ERROR()` is a new macro?  We already have `RETURN_IF_ROCBLAS_ERROR()`.

Reply:
Changed to parenthesis around rocblas_copy_template<NB, T> . The T is needed, it cannot be deduced.

Reply:
Removed CHECK_INTERNAL_ROCBLAS_ERROR macro. Added macro PRINT_AND_RETURN_IF_ROCBLAS_ERROR.